### DISCUSSION TITLE
My ball is never stopping

### DISCUSSION THREAD

**Discussion by vsirvent** [DanielChappuis/reactphysics3d#373] [Q&A]
Hello,

I don't understand why in my code the ball in the floor is never stopping, I have configured a friction coef of 0.9 in both colliders but as soon as I apply a force and I reset the forces after a small period of time the ball moves and never stops. This is my code:

        if (force.lengthSquare() > 0) {
		body->applyWorldForceAtCenterOfMass(p.body->getForce() + force);
	}else if (count-- <= 0) {
		body->resetForce();
		body->resetTorque();				
		count = INPUT_COUNTER;
	}

**Comment by DanielChappuis** [comment]
Hello. Friction alone does not generally work for spheres because there is a very small contact point and also because the sphere is not really sliding against the floor but rolling instead.

What is generally done in this situation is to simulate friction by slowing down the ball using angular (or linear) damping as described [here](https://reactphysics3d.com/usermanual.html#x1-3600010.4) in the documentation.

You can try something like this:

`body->setAngularDamping(2)`

Note that the parameter value must be in range $\[ 0, \infty \]$ where 0 means no damping.

I hope this helps.

**Reply by vsirvent** [reply]
Hi Daniel,

Thank you for the reply, that solved it. The only thing is that it also stops in the air, I could check if it's touching the floor, but I think I can live with that. 

Thank you so much for the support and your library!

**Reply by DanielChappuis** [reply]
Note that you can also use the [EventListener](https://reactphysics3d.com/usermanual.html#x1-7800015.2) of the library to receive events when contacts occur. You might want to use this to enable damping only after a contact occured with your sphere for instance.
